include/drivers/pwm.h

```diff
@@ -232,7 +231,7 @@ static inline int pwm_pin_configure_capture(const struct device *dev,
 					    pwm_capture_callback_handler_t cb,
 					    void *user_data)
 {
-	const struct pwm_driver_api *api = (struct pwm_driver_api *)dev->api;
+	const struct pwm_driver_api *api = dev->api;
```

Comment:
These type castings are not redundant. They are needed when this header file is included from a C++ source file.

Reply:
Argh, forgot about C++ :-(